### DISCUSSION TITLE
Support for agents?

### DISCUSSION THREAD

**Discussion by bendavis78** [CopilotC-Nvim/CopilotChat.nvim#256] [General]
Microsoft added support for "agents" in vscode in their [Nov 30th update](https://github.blog/changelog/2023-11-30-github-copilot-november-30th-update/#introducing-agents-in-copilot-chat-in-vs-code) last year.

![image](https://github.com/CopilotC-Nvim/CopilotChat.nvim/assets/324392/d915f88f-720d-410a-a99f-fd1df7ee86aa)

Arguably the most valuable feature here is the `@workspace` agent, which allows copilot to search the current workspace:

>  `@workspace`: This agent has knowledge about the code in your workspace and can help you navigate it by finding relevant files or classes. The `@workspace` agent uses a meta prompt to determine what information to collect from the workspace to help answer your question.

Are there any plans to add support for this in CopilotChat.nvim?

**Comment by jellydn** [comment]
That's on open issue and we have NOT implemented yet https://github.com/CopilotC-Nvim/CopilotChat.nvim/issues/5

cc @gptlang @deathbeam

**Reply by gptlang** [reply]
From what I understand, `@workspace` just takes all your code, embeds it, and uses cosine similarity to send relevant code snippets. @deathbeam seems to have already implemented something similar in https://github.com/CopilotC-Nvim/CopilotChat.nvim/blob/canary/lua/CopilotChat/context.lua

It's hard to replicate their system exactly as their javascript is completely obfuscated while also using WASM in a bunch of places.

I wonder if @github has any plans to bring Copilot Chat to vim at some point considering [copilot.vim](https://github.com/github/copilot.vim) is still actively maintained.
